File: academic-system-csharp/src/SistemaAcademico/Services/ServicoTurma.cs
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;
using SistemaAcademico.Validation;

namespace SistemaAcademico.Services;

// Equivalente à classe Java: ServicoTurma
public class ServicoTurma
{
    private readonly List<Turma> turmasCadastradas = new List<Turma>();

    public void RegistrarTurma(string codigo, string titulo, string usuarioAdmin)
    {
        if (!"ADMIN".Equals(usuarioAdmin))
        {
            throw new ExcecaoAutorizacao("Operação negada: Apenas administradores podem registrar turmas.");
        }
        Turma novaTurma = new Turma(codigo, titulo);
        ValidadorDominio.Validate(novaTurma);
        turmasCadastradas.Add(novaTurma);
    }

    public List<Turma> ListarTurmas()
    {
        return turmasCadastradas;
    }
}

File: academic-system-java/src/main/java/org/example/service/ServicoTurma.java
package org.example.service;

import java.util.ArrayList;
import java.util.List;
import org.example.model.Turma;
import org.example.exception.ExcecaoAutorizacao;
import org.example.validation.ValidadorDominio;

public class ServicoTurma {
    
    private final List<Turma> turmasCadastradas = new ArrayList<>();
    
    public void registrarTurma(String codigo, String titulo, String usuarioAdmin) {
        if (!"ADMIN".equals(usuarioAdmin)) {
            throw new ExcecaoAutorizacao("Operação negada: Apenas administradores podem registrar turmas.");
        }
        Turma novaTurma = new Turma(codigo, titulo);
        ValidadorDominio.validate(novaTurma);
        turmasCadastradas.add(novaTurma);
    }
    
    public List<Turma> listarTurmas() {
        return turmasCadastradas;
    }
}